pkg/apps/installer.go

```diff
@@ -339,7 +365,14 @@ func (i *Installer) update() error {
 		newPermissions := newManifest.Permissions()
 		samePermissions := newPermissions != nil && oldPermissions != nil &&
 			newPermissions.HasSameRules(oldPermissions)
-		if !samePermissions && !i.permissionsAcked {
+
+		// Check if we are going to skip the permissions
+		skip, err := i.checkSkipPermissions()
+		if err != nil {
+			return err
+		}
```

Comment:
It would be better to move that inside the `if !samePermissions && !i.permissionsAcked` block. There is no need to load the instance and check if we skip the permissions on this instance context in most cases (when permissions haven't changed).